fit1d.py

```diff
@@ -0,0 +1,328 @@
+
+    order : int
+        Order of function to be fitted.  Since only splines are currently supported,
+        the order is the same as the number of equal reqions the data domain is split into,
+        and 1 more than the number of equally-spaced knots.
+
+    Returns
+    -------
+
+    fit : scipy.interpolate.fitpack2.LSQUnivariateSpline object
+        This can be used to calculate the fitted values for the input x values:
+
+        fitted_values = fit(x)
+
+    """
+
+    nx = len(x)
+    if function[:6] == "spline":
+        #
+        # Number of knots is 1 fewer than the order
+        knots = get_knots(x, order-1)
+        if function == "spline3":
+            k = 3
+        elif function == "spline1":
+            k = 1
+        else:
+            print("Spline fitting only valid for spline1 and spline3 functions")
+            return None
+        fitted = LSQUnivariateSpline(x, y, knots, w=weights, k=k)
+    else:
+        print("Not implemented yet")
+        return None
+    return fitted
+
+def fit_with_rejection(x, y, weights, function, order, low_reject, high_reject, niterate, grow):
+    """Fit with sigma-clipping
+
+    Parameters
+    ----------
+
+    x : 1-d ndarray
+        The array of independent values
+
+    y : 1-d ndarray
+        The array of dependent values.  Must have same length as x, or function returns None
+
+    weights : 1-d ndarray or None
+        The array of weights.  If set to None, the points are given equal weight of 1.0.  If not
+        set to None, must have the same length as x and y, or the function returns None
+
+    function : str
+        Fitting function.  Currently only "spline1" and "spline3" are supported
+
+    order : int
+        Order of fitting fuction.  The number of knots in the spline fit is (order-1)
+
+    low_reject : float
+        Points with y values less than fit - low_reject*(rms deviation of data from fit) are rejected
+        when the fit is iterated.  Ignored if niterate = 0.
+
+    high_reject : float
+        Points with y values greater than fit + low_reject*(rms deviation of data from fit) are rejected
+        when the fit is iterated.  Ignored if niterate = 0.
+
+    niterate : int
+        Number of times sigma-clipping iterations are performed.  niterate=0 corresponds to no
+        sigma clipping
+
+    grow : float
+        Used to calculate how many elements adjacent to reject pixels are also rejected.
+        Not currently implemented, but included to duplicate the IRAF fitting parameter.
+
+    Returns
+    -------
+
+    fit : scipy.interpolate.fitpack2.LSQUnivariateSpline object
+        The result of the fit.  It can be used to calculate the fitted
+        values for the input x values:
+
+        fitted_values = fit(x)
+
+    """
+
+    fitted = fit_once(x, y, weights, function, order)
+    if niterate <= 0:
+        return fitted
+    npts = len(x)
+    rej_weights = np.ones(npts)
+    for iter in range(niterate):
```

Comment:
Best practice is probably to avoid overriding `iter`:
`for _ in range(niterate):`